src/main/kotlin/flag/FlagConfigStreamApi.kt

```diff
@@ -0,0 +1,139 @@
+package com.amplitude.experiment.flag
+
+import com.amplitude.experiment.evaluation.EvaluationFlag
+import com.amplitude.experiment.util.*
+import com.amplitude.experiment.util.SseStream
+import kotlinx.serialization.decodeFromString
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.sse.EventSource
+import okhttp3.sse.EventSourceListener
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+import java.util.concurrent.atomic.AtomicBoolean
+
+internal open class FlagConfigStreamApiError(message: String?, cause: Throwable?): Exception(message, cause) {
+    constructor(message: String?) : this(message, null)
+    constructor(cause: Throwable?) : this(cause?.toString(), cause)
+}
+internal class FlagConfigStreamApiConnTimeoutError: FlagConfigStreamApiError("Initial connection timed out")
+internal class FlagConfigStreamApiDataCorruptError: FlagConfigStreamApiError("Stream data corrupted")
+internal class FlagConfigStreamApiStreamError(cause: Throwable?): FlagConfigStreamApiError("Stream error", cause)
+
+private const val CONNECTION_TIMEOUT_MILLIS_DEFAULT = 1500L
+private const val KEEP_ALIVE_TIMEOUT_MILLIS_DEFAULT = 17000L // keep alive sends at 15s interval. 2s grace period
+private const val RECONN_INTERVAL_MILLIS_DEFAULT = 15 * 60 * 1000L
+internal class FlagConfigStreamApi (
+    deploymentKey: String,
+    serverUrl: HttpUrl,
+    httpClient: OkHttpClient = OkHttpClient(),
+    val connectionTimeoutMillis: Long = CONNECTION_TIMEOUT_MILLIS_DEFAULT,
+    keepaliveTimeoutMillis: Long = KEEP_ALIVE_TIMEOUT_MILLIS_DEFAULT,
+    reconnIntervalMillis: Long = RECONN_INTERVAL_MILLIS_DEFAULT
+) {
+    var onInitUpdate: ((List<EvaluationFlag>) -> Unit)? = null
+    var onUpdate: ((List<EvaluationFlag>) -> Unit)? = null
+    var onError: ((Exception?) -> Unit)? = null
```

Comment:
These callbacks can be directly set. Should I add setters or direct access is fine?